src/impl/peerconnection.cpp

```diff
@@ -439,24 +439,36 @@ void PeerConnection::rollbackLocalDescription() {
 	}
 }
 
-bool PeerConnection::checkFingerprint(const std::string &fingerprint) const {
+bool PeerConnection::checkFingerprint(const std::string &fingerprint, const CertificateFingerprint::Algorithm &algorithm) {
```

Comment:
You don't need to add the fingerprint to the callback, you can simply set the member variable `mRemoteFingerprintAlgorithm` in `initDtlsTransport` instead of the local one `fingerprintAlgorithm`.